client/my-sites/plugins/plugins-navigation-header/index.jsx

```diff
@@ -192,7 +186,7 @@ const PluginsNavigationHeader = ( { navigationHeaderRef, categoryName, category,
 				<UploadPluginButton
 					isMobile={ isMobile }
 					siteSlug={ selectedSite?.slug }
-					hasUploadPlugins={ hasUploadPlugins }
+					hasUploadPlugins={ selectedSite }
```

Comment:
Nit: What do you think about making more explicitly you just want a boolean flag?

```suggestion
					hasUploadPlugins={ Boolean( selectedSite ) }
```

or 


```suggestion
					hasUploadPlugins={ !!selectedSite  }
```